### PR TITLE
Make dashboard items list into a customizable variable

### PR DESCRIPTION
**Pull request by gexplorer**
I changed the **dashboard-items** variable into a customizable variable to allow dashboard items customization from within customize.

### CODE REVIEW

File: dashboard-widgets.el [hunk 1]
```diff
@@ -85,13 +85,22 @@ If the value is nil then no banner is displayed.")
...
-                          (agenda    . 5))
-  "Association list of items to show in the startup buffer.
-Will be of the form `(list-type . list-size)`.
-If nil it is disabled.  Possible values for list-type are:
-`recents' `bookmarks' `projects' `agenda' `registers'")
+(defcustom dashboard-items '((recents   . 5)
+                             (bookmarks . 5)
+                             (agenda    . 5))
+  "List of sections and items to show in the startup buffer."
+  :type '(repeat
```
Comment (JesusMtnez) [inline dashboard-widgets.el]: What happens here with custom widgets? If I'm understanding this `:type` section, it defines that that the items in the list can only be one of the first default widgets types, right?

Comment (gexplorer) [reply dashboard-widgets.el]: Yes, can you give me an example of a custom widget so I can update the type?

Comment (JesusMtnez) [reply dashboard-widgets.el]: I have an example in my own configuration: https://gitlab.com/JesusMtnez/emacs.d/blob/master/core/20-core-ui.el#L26

But everyone should be able to define their own widgets if they want.

File: dashboard-widgets.el [hunk 2]
```diff
@@ -85,13 +85,22 @@ If the value is nil then no banner is displayed.")
...
+          (cons
+           (choice
+            (const :tag "Recents" recents)
+            (const :tag "Bookmarks" bookmarks)
+            (const :tag "Agenda" agenda)
+            (const :tag "Projects" projects)
+            (const :tag "Registers" registers)
+            )
+           (integer :tag "Size")
+           ))
```
Comment (jcs090218) [inline dashboard-widgets.el:102]: Try not having brackets on the line on it's own.

**Review by jcs090218 — COMMENTED** [review]
Can you fix this for us? Thanks

**Pull request closed without merging**